marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/xquery-lib.xqy

```diff
@@ -105,3 +106,52 @@ declare function remove-empty-structured-properties($element as item()*) as docu
     }
   }
 };
+
+declare %private variable $fetch := '
+  declare variable $uri as xs:string external;
+  fn:doc($uri)
+';
+declare %private variable $put := '
+  declare variable $uri as xs:string external;
+  declare variable $node as node() external;
+  declare variable $collection as xs:string external;
+  declare variable $xslt-uri as xs:string := $uri||".xslt";
+  declare variable $options :=
+    <options xmlns="xdmp:document-insert">
+      <permissions>{xdmp:document-get-permissions($uri,"elements")}</permissions>
+      <collections>{for $c in xdmp:document-get-collections($uri) return <collection>{$c}</collection>,
+        <collection>{$collection}</collection>
+      }</collections>
+    </options>;
+
+  xdmp:document-insert($xslt-uri, $node, $options)
+';
+
+declare function function-metadata-put(
+  $uri as xs:string
+) as empty-sequence()
+{
+  let $source :=
+    xdmp:eval(
+      $fetch,
+      map:map()=>map:with("uri",$uri),
+      map:map()=>map:with("database",xdmp:modules-database()))
+  let $compiled := xquery-lib:function-metadata-compile($source)
+  return
+    xdmp:eval($put,
+      map:map()=>map:with("uri",$uri)=>
+      map:with("node",$compiled)=>
+      map:with("collection",$es:FUNCTIONDEF_COLLECTION)
+      ,
+      map:map()=>map:with("database",xdmp:modules-database()))
+};
+
+declare function function-metadata-compile(
```

Comment:
Is the only difference with these functions that they're using the DHF version of the stylesheet?

Whatever the difference is, please add comments here to explain why we've copied/tweaked these ES functions so a future developer understands the intent.

Reply:
Essentially we are using xqueryLib.functionMetadataPut instead of OOTB es.functionMetadataPut. I will add a comment both in this file and also in mlGenerateFunctionMetadata.sjs where it is used.

Reply:
Added a comment both in this file and in mlGenerateFunctionMetadata.sjs